controllers/uninstall.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2020 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	backplanev1 "github.com/stolostron/backplane-operator/api/v1"
+	"github.com/stolostron/backplane-operator/pkg/status"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var (
+	// The uninstallList is the list of all resources from previous installs to remove. Items can be removed
+	// from this list in future releases if they are sure to not exist prior to the current installer version
+	uninstallList = func(backplaneConfig *backplanev1.MultiClusterEngine) []*unstructured.Unstructured {
+		removals := []*unstructured.Unstructured{
+			newUnstructured(
+				types.NamespacedName{Name: "hypershift-deployment", Namespace: backplaneConfig.Spec.TargetNamespace},
+				schema.GroupVersionKind{Group: "", Kind: "ServiceAccount", Version: "v1"},
+			),
+			newUnstructured(
+				types.NamespacedName{Name: "hypershift-deployment-controller", Namespace: backplaneConfig.Spec.TargetNamespace},
+				schema.GroupVersionKind{Group: "apps", Kind: "Deployment", Version: "v1"},
+			),
+			newUnstructured(
+				types.NamespacedName{Name: "open-cluster-management:hypershift-preview:hypershiftDeployment-leader-election", Namespace: backplaneConfig.Spec.TargetNamespace},
+				schema.GroupVersionKind{Group: "rbac.authorization.k8s.io", Kind: "Role", Version: "v1"},
+			),
+			newUnstructured(
+				types.NamespacedName{Name: "open-cluster-management:hypershift-preview:hypershiftDeployment-leader-election", Namespace: backplaneConfig.Spec.TargetNamespace},
+				schema.GroupVersionKind{Group: "rbac.authorization.k8s.io", Kind: "RoleBinding", Version: "v1"},
+			),
+			newUnstructured(
```

Comment:
Does deleting this CRD pose a risk of data loss? We typically leak CRDs to avoid this risk.